instagram/src/components/PostContainer/PostPage.js

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from 'react';
+// import Proptypes from 'prop-types';
+
+import '../../App.scss';
+import dummyData from '../../dummy-data';
+import PostContainer from './PostContainer';
+import SearchBar from './../SearchBar/SearchBar';
+
+class PostPage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      posts: [],
+      searchQuery: '',
+
+    };
+  };
+
+  componentDidMount() {
+
+    this.setState({
+      posts: dummyData,
+    })
+
+  }
+
+  handleCommentAdd = (commentsData) => {
+    console.log('commentsData', commentsData);
+
+    const foundPostIndex = this.state.posts.findIndex((post) => {
+      return post.id === commentsData.postId;
+    });
+
+    // const clonedPosts = [...this.state.posts]
+
+    const foundPost = this.state.posts[foundPostIndex];
+    foundPost.comments = commentsData.updatedComments;
+
+    const updatedPosts = [...this.state.posts];
+
+    this.setState({
+      posts: updatedPosts,
+    });
+  }
+
+  handleLike = (likeData) => {
+    const foundPostIndex = this.state.posts.findIndex((post) => {
+      return post.id === likeData.postId;
+    });
+
+    const foundPost = this.state.posts[foundPostIndex];
+    foundPost.likes = likeData.newLikeValue;
+
+    const updatedPosts = [...this.state.posts];
+    this.setState({
+      posts: updatedPosts,
+    });
+  }
+
+  handleSearch =(searchQuery) => {
+    console.log('SEARCHQUERY:', searchQuery);
+    this.setState({searchQuery: searchQuery});
+  }
+
+  logOut = () => {
+    this.props.onLogout();
```

Comment:
you don't need to do this . It's OK to pass your onLogout straight from props without creating a method on this component